boot/freeldr/freeldr/arch/realmode/bmouse.inc

```diff
@@ -0,0 +1,166 @@
+/*
+ * PROJECT:     FreeLoader
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Bus Mouse Interrupt Detection
+ * COPYRIGHT:   Copyright 2023 Dmitry Borisov <di.sean@protonmail.com>
+ */
+
+#define PIC1_BASE         HEX(20)
```

Comment:
Question: What requires this code to be run in real mode and ASM? Couldn't it be done in C like some of our other detection routines?

Reply:
In protected mode, the IRQs 0 to 32 are reserved for CPU generated exceptions. Ok, let's hope this code won't cause any exceptions.

Reply:
You probably meant interrupt vectors, IRQs are different, usually having a more narrow range (but depending on the type of PIC).